### DISCUSSION TITLE
Load after init

### DISCUSSION THREAD

**Discussion by yjs0014** [epochpic/epoch#225] [Q&A] [answered]
Now, EPOCH code only support the situation that particles are loaded (or injected) initially. However, there are many situations (e.g. Persistent radiation produces electrons) need the fuction that particels can be loaded after init. 
Would the development team consider to add this fuction in the future?

**Comment by TomGoffrey** [comment, marked as answer]
EPOCH can load particles after t=0. The injector runs for t>0, and EPOCH can also create particles via ionisation events, QED processes and bremsstrahlung emission.

We aren't currently planning on adding any new features which add to this set, but I would think that the above routines are good starting points for anyone wanting to add similar features. How useful, and which one to use, would depend on what it was you wanted to model.

**Reply by yjs0014** [reply]
Thanks for your reply, I think I can change the module injector with moving inject boundary to accomplish this. Thanks a lot.

**Reply by TomGoffrey** [reply]
No problem.

As long as you can achieve the desired effect from the boundaries, the injectors should help.

In the past users have tried to repurpose the injectors for adding particles within the domain, but there are problems with that approach. See https://github.com/Warwick-Plasma/epoch/issues/86#issuecomment-778163097 for example.
